### PR TITLE
Better examples

### PR DESCRIPTION
**Pull request by LysandreJik**
Refactored the examples section: removed old and not up-to-date examples and added new examples for fine-tuning and generation.

The `examples` file is not in the `/doc/source` folder anymore but in the `/examples` folder. It is therefore visible when users open the folder on GitHub. 

Note: In order to generate the current documentation, a symlink has to be done between the `examples/README.md` file to a `docs/source/examples.md`. The corresponding documentation has been added to the documentation README, alongside the command necessary to create the symlink.

**The new examples are visible on: http://lysand.re/examples.html**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [Language Model fine-tuning](#language-model-fine-tuning) | Fine-tuning the library models for language modeling on a text dataset. Causal language modeling for GPT/GPT-2, masked language modeling for BERT/RoBERTa. |
+| [Language Generation](#language-generation) | Conditional text generation using the auto-regressive models of the library: GPT, GPT-2, Transformer-XL and XLNet.                                         |
+| [GLUE](#glue) | Examples running BERT/XLM/XLNet/RoBERTa on the 9 GLUE tasks. Examples feature distributed training as well as half-precision.                              |
+| [SQuAD](#squad) | Using BERT for question answering, examples with distributed training.                                                                                     |
+
+## Language model fine-tuning
+
+Based on the script `run_lm_finetuning.py`.
```
Comment (thomwolf) [inline examples/README.md]: add a link to the file?

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+a score of ~20 perplexity once fine-tuned on the dataset.
+
+### RoBERTa/BERT and masked language modeling
+
+The following example fine-tunes RoBERTa on WikiText-2. Here too, we're using the raw WikiText-2. The loss is different
+as BERT/RoBERTa have a bidirectional mechanism; we're therefore using the same loss that was used during their
+pre-training: masked language modeling. 
+
+In accordance to the RoBERTa paper, we use dynamic masking rather than static masking. The model may therefore converge
+slower, but over-fitting would take more epochs.
```
Comment (thomwolf) [inline examples/README.md]: The model may, therefore, converge slightly slower (over-fitting take more epochs).

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+Before running anyone of these GLUE tasks you should download the
+[GLUE data](https://gluebenchmark.com/tasks) by running
+[this script](https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e)
+and unpack it to some directory `$GLUE_DIR`.
+
+```bash
+export GLUE_DIR=/path/to/glue
+export TASK_NAME=MRPC
+
+python run_bert_classifier.py \
```
Comment (thomwolf) [inline examples/README.md]: Here it's `run_glue.py` now

File: examples/README.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+  --task_name $TASK_NAME \
+  --do_train \
+  --do_eval \
+  --do_lower_case \
+  --data_dir $GLUE_DIR/$TASK_NAME \
+  --bert_model bert-base-uncased \
```
Comment (thomwolf) [inline examples/README.md]: I don't think we have this argument anymore.

File: examples/README.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+Before running anyone of these GLUE tasks you should download the
+[GLUE data](https://gluebenchmark.com/tasks) by running
+[this script](https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e)
+and unpack it to some directory `$GLUE_DIR`.
+
+```bash
+export GLUE_DIR=/path/to/glue
+
+python run_bert_classifier.py \
```
Comment (thomwolf) [inline examples/README.md]: Here as well, the command line is slightly out-dated

File: examples/README.md [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+
+#### Using Apex and mixed-precision
+
+Using Apex and 16 bit precision, the fine-tuning on MRPC only takes 27 seconds. First install 
+[apex](https://github.com/NVIDIA/apex), then run the following example:
+
+```bash
+export GLUE_DIR=/path/to/glue
+
+python run_bert_classifier.py \
```
Comment (thomwolf) [inline examples/README.md]: Update

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+$SQUAD_DIR directory.
+
+* [train-v1.1.json](https://rajpurkar.github.io/SQuAD-explorer/dataset/train-v1.1.json)
+* [dev-v1.1.json](https://rajpurkar.github.io/SQuAD-explorer/dataset/dev-v1.1.json)
+* [evaluate-v1.1.py](https://github.com/allenai/bi-att-flow/blob/master/squad/evaluate-v1.1.py)
+
+```bash
+export SQUAD_DIR=/path/to/SQUAD
+
+python run_bert_squad.py \
```
Comment (thomwolf) [inline examples/README.md]: Here it's `run_squad` now

File: examples/README.md [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+  --bert_model bert-base-uncased \
```
Comment (thomwolf) [inline examples/README.md]: And we also don't have `bert_model` anymore

**Comment by LysandreJik** [comment]
There were indeed quite a few artifacts. I fixed them in the two latest commits.

**Comment by thomwolf** [comment]
Awesome! Merging

**Pull request merged**